### ISSUE TITLE
BUG: bug in outlier_test index if order is True

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#3971]
checking code for #2727
Looking at the code, I think there is a bug or order=False and the original pandas index is available. AFAICS the data is sorted but not the pandas index.

needs an example to check, because I'm not sure about the pandas behavior here.

most likely a bug in the returned index if `order` is trueish

**Comment by josef-pkt** [comment]
-BUG confirmed, we get the wrong labels in the ordered array
-another problem the OLSResults method doesn't use all keywords of the outlier_test function, e.g. order is missing.
-labels keyword in function does not have a docstring entry

this works correctly:
`res.outlier_test().sort_values(['student_resid'], ascending=False)`

the following returns an incorrect index that has not been sorted to correspond to the sorted values:
```
import statsmodels.stats.outliers_influence as oi
ot2 = oi.outlier_test(res, order=True)
```

code:
```
    if labels is None:
        labels = getattr(model_results.model.data, 'row_labels', None)
```
is too late, the labels (if user provided) are sorted earlier, but not if the labels are taken from model.data

so this works also correctly and produces the same result as sorting the results dataframe
`oi.outlier_test(res, order=True, labels=res.model.data.row_labels)`

this means we can just move the lines for accessing model row_labels to the top

**Comment by josef-pkt** [comment]
unit tests: there is only one test based on file search

stats.tests.test.test_diagnostic, test_outlier_test numbers against R
but the test uses explicitly provided labels
`res = oi.outlier_test(ndarray_mod, method='b', labels=labels, order=True)`

PR coming up later today

**Comment by josef-pkt** [comment]
more options

add options from function to method: labels, order
add cutoff option, closes #2727, see also e.g.
https://www.rdocumentation.org/packages/car/versions/2.1-6/topics/outlierTest

(should be ready to merge if green)
